platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ExportMigrationEntryImpl.java

```diff
@@ -0,0 +1,326 @@
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiPredicate;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.migration.ExportMigrationEntry;
+import org.codice.ddf.migration.MigrationException;
+import org.codice.ddf.migration.MigrationReport;
+import org.codice.ddf.migration.MigrationWarning;
+import org.codice.ddf.util.function.EBiConsumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class provides an implementation of the {@link ExportMigrationEntry}.
+ */
+public class ExportMigrationEntryImpl extends MigrationEntryImpl implements ExportMigrationEntry {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExportMigrationEntryImpl.class);
+
+    private final Map<String, ExportMigrationJavaPropertyReferencedEntryImpl> properties =
+            new HashMap<>(8);
+
+    private final AtomicReference<OutputStream> outputStream = new AtomicReference<>();
+
+    private final ExportMigrationContextImpl context;
+
+    private final Path absolutePath;
+
+    private final Throwable absolutePathError;
+
+    private final Path path;
+
+    private final File file;
+
+    private final String name;
+
+    /**
+     * Will track if store was attempted along with its result. Will be <code>null</code> until
+     * store() is attempted, at which point it will start tracking the first store() result.
+     */
+    protected Boolean stored = null;
+
+    /**
+     * Instantiates a new migration entry given a migratable context and path.
+     * <p>
+     * <i>Note:</i> In this version of the constructor, the path is either absolute or assumed to be
+     * relative to ${ddf.home}. It will also be automatically relativized to ${ddf.home}.
+     *
+     * @param context the migration context associated with this entry
+     * @param path    the path for this entry
+     * @throws IllegalArgumentException if <code>context</code> or <code>path</code> is <code>null</code>
+     */
+    protected ExportMigrationEntryImpl(ExportMigrationContextImpl context, Path path) {
+        Validate.notNull(context, "invalid null context");
+        Validate.notNull(path, "invalid null path");
+        Path apath;
+        IOException aerror;
+
+        try {
+            // make sure it is resolved against ddf.home and not the current working directory
+            apath = context.getPathUtils()
+                    .resolveAgainstDDFHome(path)
+                    .toRealPath(LinkOption.NOFOLLOW_LINKS);
+            aerror = null;
+        } catch (IOException e) {
```

Comment:
❓ I'm unclear about this reasoning. If there was a problem converting the path to a real path beneath ddf.home, wouldn't you be risking the path below here being outside ddf.home? Shouldn't this be a failure condition?

Or...I guess on lines 105-106 you'd relativize it, so you might end up with something like `${ddf.home}/etc/passwd`?

Reply:
At this point, I don't care since I am remembering that I cannot get to a real path so that later when you try to store it using the default store(), it will record an error. If you did not care about the file system as you are creating entries with names of your own for data not on disk (e.g. cfg admin migratable, profile migratable) then you would be using a different store() where you provide the data yourself.